### ISSUE TITLE
Will tflearn support CTC loss?

### ISSUE THREAD

**Issue by hn18001** [tflearn/tflearn#95]
TensorFlow r0.8 has added the implementation of CTC loss, will tflearn support CTC loss?

**Comment by aymericdamien** [comment]
Yes, it would be great to integrate it! So we can investigate how to, as it is a little it different from other traditional loss.

**Comment by madhavsund** [comment]
Anyone integrated CTC loss function to tflearn?

**Comment by shahinkl** [comment]
Any progress here? Any directions to achieve this?

**Comment by selcouthlyBlue** [comment]
Apparently, you can use your own loss functions on tflearn from this [issue](https://github.com/tflearn/tflearn/issues/72).

**Comment by selcouthlyBlue** [comment]
Tensorflow `ctc_loss` requires sparse labels. So, in order to integrate `ctc_loss` into tflearn, the labels must be converted within the function. Here's what I'd put in `objectives.py`:

```
def ctc_loss(y_pred, y_true, eos_token=0, sequence_length=320):
    with tf.name_scope("CTCLoss"):
        #TODO: I've created a dense_to_sparse op in tensorflow so this should be replaced once it's released.
        indices = tf.where(tf.not_equal(y_true, tf.constant(eos_token, dtype=y_true.dtype)))
        values = tf.gather_nd(y_true, indices)
        shape = tf.shape(y_true, out_type=tf.int64)
        sparse_y_true = tf.SparseTensor(
            indices,
            values,
            shape
        )
        return tf.nn.ctc_loss(labels=sparse_y_true, inputs=y_pred, sequence_length=sequence_length)
```

The only problem is passing a dynamic `sequence_length`.

**Comment by selcouthlyBlue** [comment]
I have already made a pull request https://github.com/tflearn/tflearn/pull/1008
